cl/users/urls.py

```diff
@@ -102,6 +102,21 @@
     path("profile/api-token/", views.view_api_token, name="view_api_token"),
     path("profile/api-usage/", views.view_api_usage, name="view_api_usage"),
     path("profile/webhooks/", views.view_webhooks, name="view_webhooks"),
+    path(
+        "profile/webhook-form/",
+        views.webhook_form,
+        name="webhook_form",
+    ),
+    path(
+        "profile/webhooks-list/",
+        views.view_webhooks_list,
+        name="view_webhooks_list",
+    ),
+    path(
+        "profile/webhook-delete/<int:pk>/",
+        views.delete_webhook,
+        name="delete_webhook",
```

Comment:
I just read many tutorials on HTMX and Django, and I don't like any of them. They all take the approach you did here, which uses multiple views separating out logic. Doing this leaves our views untethered to each other and spreads our code out very thinly. 

I think a better way to do this is to do as few views as possible, and to use HTTP verbs. So:

 - HTTP DELETE sent to /api/rest/v3/webhooks/<id>/ deletes something
 - HTTP PUT sent to /api/rest/v3/webhooks/<id>/ updates something
 - HTTP POST sent to /api/rest/v3/webhooks/ creates something
 - HTTP GET sent to /api/rest/v3/webhooks/ lists them
 - HTTP GET sent to /api/rest/v3/webhooks/<id> lists one (if needed)

This mostly mirrors the approach in Django REST Framework, and you can see that I used its URLs in my above concept. 

I spent some time looking for an integration between DRF and HTMX, but didn't find anything good. It seems like it should exist, but I guess not (yet). 

In any case, I think we should do something like the above. If we can use DRF to power HTMX, we should do that, using the URLs above. If we can't, we should use HTTP verbs and as few URL paths as possible, probably putting them at `/webhooks/` instead of under `/api/rest/v3/`.

Does this make sense? Do you agree?

Reply:
The other thing that might be worth doing is to use django-htmx to combine views. For example, you have two that basically just list webhooks. Those could be combined by using the [django-htmx middleware](https://django-htmx.readthedocs.io/en/latest/middleware.html#django_htmx.middleware.HtmxDetails.__bool__).

Reply:
Thanks! this is a better approach to implementing htmx with django, I managed to use DRF to render over an HTML template (needed for htmx) using a ModelViewSet I could implement all the necessary actions for creating, editing, updating, and deleting webhooks.

Now it's possible to handle all the actions using one route: `/api/rest/v3/webhooks` and different verbs such as DELETE, PUT, POST, and GET.

In order to use `/api/rest/v3/webhooks` added this router on `cl.api.urls `, however, there is something that makes me think maybe that's wrong.  Because if you go to `/api/rest/v3/` it's possible to see the webhooks endpoint but this is not a normal API endpoint due to its actions render to an HTML template as default, would it be better to move it out of `/api/rest/v3/`?

Reply:
Yes, I agree, now that I see it in /api/rest/v3/webhooks, it feels wrong. Let's give it a better home. A few possibilities:

 - /api/htmx/webhooks/  - I kind of like this.
 - /profile/webhooks/ - Feels limited, but maybe it's fine?
 - /webhooks/ - This feels like it's too high in the hierarchy.

What do you think?

Reply:
I think this URL tweak is still needed.

Reply:
Sorry, I missed this comment. I've changed it to: `/api/htmx/webhooks/ ` I think it describes pretty well that it belongs to htmx actions.

I moved it into users.urls